File: frontend/src/api/portfolio.ts
import { apiClient } from '@/lib/apiClient';

export interface PortfolioItem {
  id: string;
  stock_id: string;
  symbol: string;
  quantity: number;
  average_price: number;
  current_price?: number;
  pnl?: number;
  pnlPercent?: number;
}

export interface PortfolioSummary {
  totalValue: number;
  totalPnl: number;
  totalPnlPercent: number;
  riskExposure: 'LOW' | 'MEDIUM' | 'HIGH';
}

export const portfolioApi = {
  getPortfolio: () =>
    apiClient.get<PortfolioItem[]>('/portfolio/'),

  getRiskSummary: () =>
    apiClient.get<PortfolioSummary>('/portfolio/risk-summary'),
};

File: backend/app/api/portfolio.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from app.database.session import get_db
from app.models.base import Portfolio, Stock, AILog

router = APIRouter()

@router.get("/", summary="Get user portfolio positions")
def get_portfolio(db: Session = Depends(get_db)):
    """Returns real portfolio data from PostgreSQL."""
    # TODO: In production, filter by JWT user_id
    portfolios = db.query(Portfolio).filter(Portfolio.is_deleted == False).all()
    return [
        {
            "id": str(p.id),
            "stock_id": p.stock_id,
            "quantity": p.quantity,
            "average_price": p.average_price,
        }
        for p in portfolios
    ]

@router.post("/trade", summary="Add a trade to portfolio")
def create_trade(db: Session = Depends(get_db)):
    return {"message": "Trade added"}

@router.get("/risk-summary", summary="Get portfolio risk summary")
def risk_summary(db: Session = Depends(get_db)):
    portfolios = db.query(Portfolio).filter(Portfolio.is_deleted == False).all()
    total_value = sum(p.quantity * p.average_price for p in portfolios)
    return {
        "totalValue": total_value,
        "totalPnl": 0,
        "totalPnlPercent": 0,
        "riskExposure": "LOW",
    }
